### PR TITLE
HHH-12935 : Constraint and AuxiliaryDatabaseObject export identifiers are not qualified by schema or catalog

### PR DESCRIPTION
**Pull request by gbadner**
https://hibernate.atlassian.net/browse/HHH-12935

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/test/schemaupdate/ExportIdentifierTest.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+		try {
+			addTables( "aTable" , database.getNamespaces(), exportIdentifierList, exportIdentifierSet );
+			addSimpleAuxiliaryDatabaseObject( database.getNamespaces(), exportIdentifierList, exportIdentifierSet );
+			addNamedAuxiliaryDatabaseObjects(
+					"aNamedAuxiliaryDatabaseObject", database.getNamespaces(), exportIdentifierList, exportIdentifierSet
+			);
+			addSequences( "aSequence", database.getNamespaces(), exportIdentifierList, exportIdentifierSet );
+
+			assertEquals( exportIdentifierList.size(), exportIdentifierSet.size() );
```
[the remark this answers is not in the document]
Comment (gsmet) [reply hibernate-core/src/test/java/org/hibernate/test/schemaupdate/ExportIdentifierTest.java]: Withdrawing this comment as the quoting would be different for each RDBMS.

File: hibernate-core/src/main/java/org/hibernate/boot/model/relational/SimpleAuxiliaryDatabaseObject.java [hunk 2]
```diff
@@ -93,6 +93,14 @@ public SimpleAuxiliaryDatabaseObject(
...
 	}
 
+	protected String getCatalogName() {
+		return catalogName;
+	}
+
+	protected String getSchemaName() {
+		return schemaName;
+	}
+
```
[the remark this answers is not in the document]
Comment (gbadner) [reply hibernate-core/src/main/java/org/hibernate/boot/model/relational/SimpleAuxiliaryDatabaseObject.java]: The export identifier is intended to ensure that the same statement doesn't get exported more than once, so uniqueness is the most important thing.

SimpleAuxiliaryDatabaseObject doesn't actually have a name. The superclass uses a counter [1], so it is already unique. That's why I didn't bother with prefixing with catalog or schema.

[1] https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/boot/model/relational/AbstractAuxiliaryDatabaseObject.java#L49

**Comment by gsmet** [comment]
@gbadner I think we can merge this one?

**Comment by gbadner** [comment]
Merged.

**Pull request closed without merging**